### DISCUSSION TITLE
Aspire and projects with existing IaC (bicep)

### DISCUSSION THREAD

**Discussion by jedjohan** [microsoft/aspire#5105] [Q&A] [answered]
We have extensive IaC using bicep (main.bicep + [ENV].parameters.json). The main.bicep is using bicep modules that we store in ACR. The infra setup include all resources (databases, storage accounts and whatever the app need) + cross subscription permissions using managed identity + tagging of all resources + config in azure app configuration and is executed in AZDO infra-pipeline.

What would be a good path to introduce aspire for such an existing setup ?

**Comment by davidfowl** [comment]
You could use aspire solely for local development and continue deploying the way you have been (using custom bicep). The way connection strings are provided by various aspire resources would be the only thing you'd need to align between local development and deployment (how and what configuration is provided to the application)

**Reply by jedjohan** [reply]
Ok thanks. But then we don't get the aspire dashboard in azure I guess, which is kind of what we are after as we already have the infra in quite a mature state.

**Reply by davidfowl** [reply, marked as answer]
You can turn on the aspire dashboard in ACA by tweaking your existing bicep (or by turning it on in the portal). You'll need to emit otel from your applications though.

**Reply by jedjohan** [reply]
(Just curious, how did you know we are running ACA 🧐)
